cloud/docker/docker_container.py

```diff
@@ -1185,11 +1184,10 @@ def has_different_configuration(self, image):
             privileged=host_config.get('Privileged'),
             expected_ports=host_config.get('PortBindings'),
             read_only=host_config.get('ReadonlyRootfs'),
-            restart_policy=restart_policy.get('Name'),
-            restart_retries=restart_policy.get('MaximumRetryCount'),
+            restart_policy=host_config.get('RestartPolicy'),
```

Comment:
This is not correct. The value in self.parameter.restart_policy is of type string, and not a dict. 

See line 705. That is where restart_policy is defined. It will receive the value supplied by the task parameters, if any. At that point it is a property of a TaskParameters object, which gets passed into the Container object and used in the has_different_configuration() method.